### ISSUE TITLE
Resign command doesn't filter builds by profile

### ISSUE THREAD

**Issue by d4rky-pl** [expo/eas-cli#2404]
### Summary

When running `eas build:resign --platform ios --profile testing` the interactive list of builds includes builds from all profiles, not just from `testing`

### Managed or bare?

managed

### Environment

```
  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.4
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 16.15.0 - ~/.asdf/installs/nodejs/16.15.0/bin/node
      Yarn: 1.22.15 - ~/.asdf/installs/nodejs/16.15.0/bin/yarn
      npm: 8.5.5 - ~/.asdf/plugins/nodejs/shims/npm
      Watchman: 2024.05.06.00 - /opt/homebrew/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
      Android SDK:
        API Levels: 33, 33, 34
        Build Tools: 30.0.3, 33.0.0, 33.0.2, 34.0.0, 34.0.0
        System Images: android-33 | Google APIs ARM 64 v8a, android-34 | Google APIs ARM 64 v8a
    IDEs:
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      @expo/webpack-config: ^18.0.1 => 18.0.2
      babel-preset-expo: ^9.3.0 => 9.3.1
      expo: ^48.0.0 => 48.0.9
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.71.4 => 0.71.4
      react-native-web: ~0.18.11 => 0.18.12
    npmGlobalPackages:
      eas-cli: 9.1.0
    Expo Workflow: managed
```

### Reproducible demo or steps to reproduce from a blank project

1. Create a project with `profileA` and `profileB` in eas.json
2. Build iOS apps for both profiles
3. Run `eas build --platform ios --profile profileA`

**Comment by d4rky-pl** [comment]
@szdziedzic I mean `build:resign`, my bad. There's no --buildProfile flag according to the documentation

(sorry for closing the ticket, butterfingers 😅)

**Comment by szdziedzic** [comment]
Oh, sorry, now I get it. Yeah, you are right, thanks for report!

**Comment by szdziedzic** [comment]
@d4rky-pl Oh, actually, I believe the name of the `--profile` flag is quite unfortunate and this might be an unwanted behavior. 

What I mean is that `--profile` is actually a **target profile**, so it defines a profile for which you want to use credentials and env vars to resign some other eligible build.  I believe that a use case for some people can be resigning build originally created using profile A using credentials and env vars from profile B that may differ. So this new filtering by the `--profile` flag might be unwanted in their use case.

**Comment by szdziedzic** [comment]
What I think we can do here is to create 2 new flags:
- `target-profile` which will do what `profile` does
- `source-profile` for filtering

What do you think?

**Comment by d4rky-pl** [comment]
@szdziedzic after hearing your explanation I think it makes a lot of sense, I didn't realize there could be a use-case like these as we use the build profiles for specific environments (`staging`, `production` etc) so we would never want to mix ENV vars or credentials this way

**Comment by szdziedzic** [comment]
`--target-profile` and `source-profile` flags will be released in `9.2.0`
